src/libstore/s3_binary_cache_store.go

```diff
@@ -0,0 +1,131 @@
+package libstore
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/minio/minio-go/v7"
+	"io"
+	"log"
+	"net/url"
+	"os"
+)
+
+type S3Store struct {
+	Profile    string
+	Region     string
+	Scheme     string
+	bucketName string
+	client     *minio.Client
+}
+
+// SDK should be in environment
+func (s *S3Store) getUri() string {
+	return "s3://" + s.bucketName
+}
+
+func (s *S3Store) FileExists(ctx context.Context, path string) (bool, error) {
+	found, err := s.client.BucketExists(ctx.Background(), s.bucketName)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
```

Comment:
I wouldn't put that check here for two reasons:
* it doubles the number of requests that are issued when FileExists is called, which makes the function slower
* when GetObject is called on a bucket that doesn't exist, it will already return an error